TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/DqtNoteAttachment.cs

```diff
@@ -0,0 +1,50 @@
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace TeachingRecordSystem.Core.Dqt;
+
+public class DqtNoteAttachment : IDqtNoteAttachment
+{
+    private const string StorageContainer = "dqt-note-attachments";
+    private BlobContainerClient _blobContainerClient { get; init; }
+    private BlobServiceClient _blobServiceClient { get; init; }
+
+    public DqtNoteAttachment(BlobServiceClient blobServiceClient)
+    {
+        _blobContainerClient = blobServiceClient.GetBlobContainerClient(StorageContainer);
+        _blobServiceClient = blobServiceClient;
+    }
+
+    public async Task<bool> CreateAttachmentAsync(byte[] attachmentBytes, string fileName, string? mimetype = null)
+    {
+        var httpHeaders = new BlobHttpHeaders()
+        {
+            ContentType = mimetype
+        };
+        var blobClient = _blobContainerClient.GetBlobClient(fileName);
+        using (MemoryStream stream = new MemoryStream(attachmentBytes))
+        {
+            await blobClient.UploadAsync(stream, httpHeaders);
+        }
```

Comment:
There's an overload of `Upload` that takes a `BinaryData`; `BinaryData.FromBytes` will save allocating a `MemoryStream`.